app/src/test/java/org/simple/clinic/sync/DataSyncTest.kt

```diff
@@ -687,6 +691,151 @@ class DataSyncTest {
+
+    val purgeOnSync = mock<PurgeOnSync>()
+    val dataSync = DataSync(
+        modelSyncs = arrayListOf(modelSync1, modelSync2, modelSync3),
+        crashReporter = mock(),
+        userSession = userSession,
+        schedulersProvider = schedulersProvider,
+        syncScheduler = schedulersProvider.io(),
+        purgeOnSync = purgeOnSync
+    )
+
+    // when
+    dataSync.sync(SyncGroup.FREQUENT)
+
+    // then
+    verifyZeroInteractions(purgeOnSync)
+  }
+
+  @Test
+  fun `when syncing everything, the unused data must be purged if all syncs completed successfully`() {
+    // given
+    val modelSync1 = FakeModelSync(
+        _name = "sync1",
+        config = frequentSyncConfig
+    )
+
+    val modelSync2 = FakeModelSync(
+        _name = "sync2",
+        config = frequentSyncConfig
+    )
+
+    val modelSync3 = FakeModelSync(
+        _name = "sync3",
+        config = dailySyncConfig
+    )
+
+    val purgeOnSync = mock<PurgeOnSync>()
+    val dataSync = DataSync(
+        modelSyncs = arrayListOf(modelSync1, modelSync2, modelSync3),
+        crashReporter = mock(),
+        userSession = userSession,
+        schedulersProvider = schedulersProvider,
+        syncScheduler = schedulersProvider.io(),
+        purgeOnSync = purgeOnSync
+    )
+
+    // when
+    dataSync.syncTheWorld()
+
+    // then
+    verify(purgeOnSync).purgeUnusedData()
+    verifyNoMoreInteractions(purgeOnSync)
+  }
+
+  @Test
+  fun `if a pull operation fails when syncing everything, the unused data must not be purged`() {
+    // given
+    val modelSync1 = FakeModelSync(
+        _name = "sync1",
+        config = frequentSyncConfig,
+        pullError = RuntimeException()
+    )
+
+    val modelSync2 = FakeModelSync(
+        _name = "sync2",
+        config = frequentSyncConfig
+    )
+
+    val modelSync3 = FakeModelSync(
+        _name = "sync3",
+        config = dailySyncConfig
+    )
+
+    val purgeOnSync = mock<PurgeOnSync>()
+    val dataSync = DataSync(
+        modelSyncs = arrayListOf(modelSync1, modelSync2, modelSync3),
+        crashReporter = mock(),
+        userSession = userSession,
+        schedulersProvider = schedulersProvider,
+        syncScheduler = schedulersProvider.io(),
+        purgeOnSync = purgeOnSync
+    )
+
+    // when
+    dataSync.syncTheWorld()
+
+    // then
+    verify(purgeOnSync, never()).purgeUnusedData()
+    verifyNoMoreInteractions(purgeOnSync)
+  }
+
+  @Test
+  fun `if a push operation fails when syncing everything, the unused data must not be purged`() {
+    // given
+    val modelSync1 = FakeModelSync(
+        _name = "sync1",
+        config = frequentSyncConfig
+    )
+
+    val modelSync2 = FakeModelSync(
+        _name = "sync2",
+        config = frequentSyncConfig
+    )
+
+    val modelSync3 = FakeModelSync(
+        _name = "sync3",
+        config = dailySyncConfig,
+        pushError = RuntimeException()
+    )
+
+    val purgeOnSync = mock<PurgeOnSync>()
+    val dataSync = DataSync(
```

Comment:
Can we make `modelSync1` , `modelSync2` , `modelSync3`, `purgeOnSync` , `dataSync` as global variables? We're using them in most of the tests?

Reply:
The `modelSync` instances are not the same in the tests. They have the same names, but they behave differently.